rq/queue.py

```diff
@@ -663,11 +663,23 @@ def enqueue_call(
             on_success=on_success,
             on_failure=on_failure,
         )
+        return self._check_dependencies(job, pipeline=pipeline, at_front=at_front)
 
+    def _check_dependencies(self, job: 'Job', pipeline: Optional['Pipeline'] = None, at_front: bool = False) -> Job:
+        """Check dependencies and enqueues a job for delayed execution.
+
+        Args:
+            job (Job): The job to enqueue
+            pipeline (Optional[Pipeline], optional): The Redis pipeline to use. Defaults to None.
+            at_front (bool, optional): Whether should enqueue at the front of the queue. Defaults to False.
+
+        Returns:
+            Job: The enqued job
+        """
         job = self.setup_dependencies(job, pipeline=pipeline)
         # If we do not depend on an unfinished job, enqueue the job.
         if job.get_status(refresh=False) != JobStatus.DEFERRED:
```

Comment:
I think we can simply move these checks to `enqueue_job` and that would solve the issue (`enqueue_job` should also setup dependencies), right?

Reply:
Do you mean the `enqueue_job` added in this pull request or the original `enqueue_job`?

Reply:
I meant we should just add dependency checks to the original [enqueue_job()](https://github.com/rq/rq/blob/e92682c83aefda9475eba3dd9fe0ef42575d92b7/rq/queue.py#L982) method. So it looks something like this:
```python
def enqueue_job(self, job):
     job = self.setup_dependencies(job, pipeline=pipeline)
     # If we do not depend on an unfinished job, enqueue the job.
     if job.get_status(refresh=False) != JobStatus.DEFERRED:
        return self._enqueue_job(job, pipeline=pipeline, at_front=at_front)
    return job
```

Reply:
I extracted the code from `enqueue_call()` to `_enqueue_checking_dependencies()` for use in `enqueue_call()` and `enqueue_job()`. Would you rather I duplicate the code?

Reply:
Just move the logic in `_enqueue_checking_dependencies()` to `enqueue_job()`. `enqueue_call()` can then call `enqueue_job()`

Reply:
I will do it. Thanks.

Reply:
Done via https://github.com/rq/rq/compare/3f5e188fb93be91e9c7452d1a5724cdff9d5096a..4cf0db1ea9f2c93bbd2557882438efff3d2b77de